lib/health_cards/qbp_client.rb

```diff
@@ -0,0 +1,217 @@
+# frozen_string_literal: true
+
+require 'savon'
+require 'ruby-hl7'
+require 'faraday'
+require_relative 'qpd'
+
+module HealthCards
+  # Send, receive, and translate HL7 V2 messages from the QBP client of the IIS sandbox
+  module QBPClient # rubocop:disable Metrics/ModuleLength # Disabling since module will be ported and refactored
+    extend self
+    # Query a patient's Immunization history from the IIS Sandbox
+    # @param patient_info [Hash] Patient demographic info sent from the IIS Consumer Portal
+    # @param credentials [Hash] (optional) User Id, Password, and Facility Id for the IIS Sandbox login
+    # @return [HL7::Message] The HL7 V2 Response message from the IIS-Sandbox
+    def query(patient_info,
+              sandbox_credentials = { username: Rails.application.config.username,
```

Comment:
I've been thinking a bit about making QBP client a class. I'm currently moving this PR over to the [iis-shc-issuer](https://github.com/dvci/iis-shc-issuer) repository, so adding Rails as a dependency to our gem should no longer be an issue.

That being said, if you still think this would be suited better as a class that takes in the config settings, I would be happy to re-open this issue in that repo for a potential refactor.